### PR TITLE
implement java sdk

### PR DESCRIPTION
**Pull request by Thetitr**
https://gmslabs.atlassian.net/browse/HYB-73

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,8 +8,23 @@
     <artifactId>Hyber-SDK-Java</artifactId>
     <version>1.0.0</version>
 
+    <properties>
```
Comment (segabriel) [inline pom.xml]: remove all checkstyle (dependencies & properties)

File: src/main/java/com/hyber/domain/hyberrequest/PushOptions.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    public String getAction() {
+        return action;
+    }
+
+    private void setAction(String action) {
+        this.action = action;
+    }
+
+    public static PushOptions of(ChannelPush push) {
```
Comment (segabriel) [inline src/main/java/com/hyber/domain/hyberrequest/PushOptions.java]: A public static method must be above no static and private methods

File: src/main/java/com/hyber/domain/sdkmessage/Message.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    private Boolean isPromotional;
+    private Long startTime;
+    private Channel[] channels;
+
+
+    public Message(Long phoneNumber) {
+        this.phoneNumber = phoneNumber;
+    }
+
+    public Long getPhoneNumber() {
```
Comment (segabriel) [inline src/main/java/com/hyber/domain/sdkmessage/Message.java]: All trash like getter and setter must be at the bottom

File: src/main/java/com/hyber/send/HttpSender.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.hyber.send;
+
+import com.hyber.constants.HyberConstants;
+
+public abstract class HttpSender implements Sender {
+
+    protected String urlPattern = HyberConstants.URL_PATTERN;
+    protected Integer connectionTimeout = HyberConstants.DEFAULT_CONNECTION_TIMEOUT;
```
Comment (segabriel) [inline src/main/java/com/hyber/send/HttpSender.java]: This field has public getter and setter, thus why is it protected?

File: src/main/java/com/hyber/send/Response.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package com.hyber.send;
+
+public abstract class Response {
```
Comment (Thetitr) [inline src/main/java/com/hyber/send/Response.java]: i like this solution too, but we need accepting by @yanchenkoao

File: src/main/java/com/hyber/service/ServiceExample.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+         * If your data is malformed, exception will be thrown. Otherwise, message will be sent
+         */
+        try {
+            Response response = sender.send(message);
+
+            /**
+             * You may receive SuccessResponse or ErrorResponse
+             */
+
+            if (response instanceof SuccessResponse) {
```
Comment (Thetitr) [inline src/main/java/com/hyber/service/ServiceExample.java]: i like this solution too, but we need accepting by @yanchenkoao

File: src/main/java/com/hyber/constants/HyberConstants.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.hyber.constants;
+
+public final class HyberConstants {
+
+    public static final String URL_PATTERN = "http://185.46.89.20/http-receiver/api/%d/messages/create";
```
Comment (OlegShevchuk) [inline src/main/java/com/hyber/constants/HyberConstants.java]: Change url

Comment (yanchenkoao) [inline src/main/java/com/hyber/constants/HyberConstants.java]: set please production url

File: src/main/java/com/hyber/exception/ValidationErrors.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package com.hyber.exception;
+
+public enum ValidationErrors {
+
+    PHONE_NUMBER_INCORRECT("Phone number incorrect!"),
+    CHANNELS_IS_EMPTY("Channels can't be empty"),
+    CHANNEL_TEXT_IS_NULL("Text for channel %s can't be null!"),
+    NOT_UNIQUE_CHANNELS("Channels must be unique"),
+    TTL_NOT_A_POSITIVE_NUMBER("Ttl must be > 0");
```
Comment (yanchenkoao) [inline src/main/java/com/hyber/exception/ValidationErrors.java]: maybe between 15 and 86400?

File: src/main/java/com/hyber/send/MessageSender.java [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+    public Integer getIdentifier() {
+        return identifier;
+    }
+
+    public String getAlphaName() {
+        return alphaName;
+    }
+
+    public Response send(Message message) throws Exception {
```
Comment (yanchenkoao) [inline src/main/java/com/hyber/send/MessageSender.java]: maybe should to add annotation @Override

File: src/main/java/com/hyber/service/ServiceExample.java [hunk 10]
```diff
@@ -0,0 +1,222 @@
+package com.hyber.service;
```
Comment (yanchenkoao) [inline src/main/java/com/hyber/service/ServiceExample.java]: please remove class

File: src/main/java/com/hyber/exception/ValidationErrors.java (hunk 8 above)
Comment (Thetitr) [inline src/main/java/com/hyber/exception/ValidationErrors.java]: No. We decide make just basic validation there. What if we change 15 and 86400, we will update all instances of sdk?

**Pull request merged**